go/services/cluster_metadata/pkg/services/metadata_service.go

```diff
@@ -1,188 +1,340 @@
 package services
 
 import (
-	"fmt"
-	"os"
+	"context"
 	"slices"
-	"strconv"
 	"time"
 
 	sharedrepo "github.com/Magpie-Monitor/magpie-monitor/pkg/repositories"
 	"github.com/Magpie-Monitor/magpie-monitor/services/cluster_metadata/pkg/repositories"
 	"go.mongodb.org/mongo-driver/bson"
+	"go.uber.org/fx"
 	"go.uber.org/zap"
 )
 
-func NewMetadataService(log *zap.Logger, clusterRepo *sharedrepo.MongoDbCollection[repositories.ClusterState], nodeRepo *sharedrepo.MongoDbCollection[repositories.NodeState]) *MetadataService {
-	clusterActivityWindowMillis, present := os.LookupEnv("CLUSTER_METADATA_SERVICE_CLUSTER_ACTIVITY_WINDOW_MILLIS")
-	if !present {
-		panic("env variable CLUSTER_METADATA_SERVICE_CLUSTER_ACTIVITY_WINDOW_MILLIS not set")
-	}
+const (
+	clusterAggregatedStateUpdateSleepSeconds = 30        // how often cluster state is refreshed
+	nodeActivityWindowMillis                 = 3_600_000 // cluster node is considered as running if it has reported in this period
+	clusterActivityWindowMillis              = 3_600_000 // cluster is considered as running if it has reported in this period
+)
 
-	window, err := strconv.ParseInt(clusterActivityWindowMillis, 10, 64)
-	if err != nil {
-		panic("invalid value for env variable CLUSTER_METADATA_SERVICE_CLUSTER_ACTIVITY_WINDOW_MILLIS, please make sure it's numeric")
+func NewMetadataService(lc fx.Lifecycle, log *zap.Logger, clusterRepo *sharedrepo.MongoDbCollection[repositories.ApplicationState], nodeRepo *sharedrepo.MongoDbCollection[repositories.NodeState],
```

Comment:
Consider using wrapper of arguments for this method. Something like MetadataServiceParams.